### ISSUE TITLE
Error when disabling a button in v2.1.0.4009 === AttributeGraph: cycle detected through attribute 165912 ===

### ISSUE THREAD

**Issue by BigMacAdmin** [swiftDialog/swiftDialog#215]
I got a long string of text into standard output on the FIRST command I ran using the preview version: 2.1.0.4009

I started with:
`dialog --button1text "Ya sure okay buddy" -p & `

Then I ran this:
`echo "button1: disable" >> /var/tmp/dialog.log`

and I was deluged with the following output:
```
=== AttributeGraph: cycle detected through attribute 165912 ===
=== AttributeGraph: cycle detected through attribute 166120 ===
=== AttributeGraph: cycle detected through attribute 166120 ===
=== AttributeGraph: cycle detected through attribute 164920 ===
=== AttributeGraph: cycle detected through attribute 166572 ===
=== AttributeGraph: cycle detected through attribute 168312 ===
=== AttributeGraph: cycle detected through attribute 164920 ===
=== AttributeGraph: cycle detected through attribute 168312 ===
=== AttributeGraph: cycle detected through attribute 174180 ===
=== AttributeGraph: cycle detected through attribute 174180 ===
=== AttributeGraph: cycle detected through attribute 174180 ===
=== AttributeGraph: cycle detected through attribute 175020 ===
=== AttributeGraph: cycle detected through attribute 175020 ===
=== AttributeGraph: cycle detected through attribute 175020 ===
=== AttributeGraph: cycle detected through attribute 175020 ===
=== AttributeGraph: cycle detected through attribute 174180 ===
=== AttributeGraph: cycle detected through attribute 175020 ===
=== AttributeGraph: cycle detected through attribute 175020 ===
=== AttributeGraph: cycle detected through attribute 175020 ===
=== AttributeGraph: cycle detected through attribute 175020 ===
=== AttributeGraph: cycle detected through attribute 174180 ===
=== AttributeGraph: cycle detected through attribute 175020 ===
=== AttributeGraph: cycle detected through attribute 174696 ===
=== AttributeGraph: cycle detected through attribute 174696 ===
=== AttributeGraph: cycle detected through attribute 178424 ===
=== AttributeGraph: cycle detected through attribute 178424 ===
=== AttributeGraph: cycle detected through attribute 178424 ===
=== AttributeGraph: cycle detected through attribute 178424 ===
=== AttributeGraph: cycle detected through attribute 178424 ===
=== AttributeGraph: cycle detected through attribute 191072 ===
=== AttributeGraph: cycle detected through attribute 191224 ===
=== AttributeGraph: cycle detected through attribute 191224 ===
=== AttributeGraph: cycle detected through attribute 191376 ===
=== AttributeGraph: cycle detected through attribute 191376 ===
=== AttributeGraph: cycle detected through attribute 191808 ===
=== AttributeGraph: cycle detected through attribute 191808 ===
=== AttributeGraph: cycle detected through attribute 253704 ===
=== AttributeGraph: cycle detected through attribute 253704 ===
=== AttributeGraph: cycle detected through attribute 253704 ===
=== AttributeGraph: cycle detected through attribute 253704 ===
=== AttributeGraph: cycle detected through attribute 253704 ===
=== AttributeGraph: cycle detected through attribute 191072 ===
=== AttributeGraph: cycle detected through attribute 191224 ===
=== AttributeGraph: cycle detected through attribute 191224 ===
=== AttributeGraph: cycle detected through attribute 191376 ===
=== AttributeGraph: cycle detected through attribute 191376 ===
=== AttributeGraph: cycle detected through attribute 191808 ===
=== AttributeGraph: cycle detected through attribute 191808 ===
=== AttributeGraph: cycle detected through attribute 191224 ===
=== AttributeGraph: cycle detected through attribute 191224 ===
=== AttributeGraph: cycle detected through attribute 191376 ===
=== AttributeGraph: cycle detected through attribute 191376 ===
=== AttributeGraph: cycle detected through attribute 191808 ===
=== AttributeGraph: cycle detected through attribute 191808 ===
=== AttributeGraph: cycle detected through attribute 168312 ===
```

The problem is somewhat intermittent. but  I can reproduce the issue multiple times in a row. The attribute numbers are identical each time the problem occurs.

**Comment by BigMacAdmin** [comment]
Additional important note: The second command "hangs" in Terminal, it does not exit as expected.

**Comment by bartreardon** [comment]
weird - I can't replicate it here - you using Terminal or some other terminal app?

**Comment by BigMacAdmin** [comment]
I'm using macOS Terminal, Ventura 13.1. Nothing special regarding my setup. I'll try again tomorrow after reboots.

**Comment by bartreardon** [comment]
ok - replicated and semi-related to the `launchctl asuser` change but I understand what is happening here.

You _may_ see it still but I'd like to see what conditions it occurs under. What it's complaining about is you started a dialog with a default selected button (button1) which is also the only button or selectable thing on the window. When you disable it, it cause SwiftUI to go into a spasm complaining about there being no default button.

Interested if you can look at the release version of [2.3](https://github.com/bartreardon/swiftDialog/releases/tag/v2.3.0) and let me know if it's resolved for you or if the can replicate it under other conditions. Either way, the messages are benign and not a _huge_ amount I can do about it depending on the context it happens in.

**Comment by bartreardon** [comment]
should be fixed in 2.3.1 - can re-open if this is still an issue
